app/models/workarea/catalog/product.decorator

```diff
@@ -2,6 +2,15 @@ module Workarea
   decorate Catalog::Product, with: 'affirm' do
     decorated do
       field :affirm_available, type: Boolean, default: true
+
+      before_validation :prevent_affirm_for_gift_cards,
+        if: -> { respond_to?(:gift_card?) }
+    end
+
+    private
+
+    def prevent_affirm_for_gift_cards
+      self.affirm_available = !gift_card? unless persisted? && affirm_available_changed?
```

Comment:
This is really hard to read and seems like it could change the value for non-gift-card products.

How about 

```
def prevent_affirm_for_gift_cards
  self.affirm_available = false if gift_card? && affirm_available?
end
```

Reply:
> seems like it could change the value for non-gift-card products

Preventing this (I caught it in tests) is actually why the method is so hard to read. Originally it was just:

```ruby
self.affirm_available = !gift_card?
```

But that was messing stuff up when you tried to actually set `affirm_available:` on a product.

Your version is a bit easier to read, so I think I'll go with that.